RestApi/Sepes.RestApi/Controllers/AuthController.cs

```diff
@@ -13,6 +23,102 @@ namespace Sepes.RestApi.Controller
+            
+            return response;
+        }
+        //Takes old token, and generates a new token.
+        [Authorize]
+        [HttpPost("refreshtoken")]
+        public IActionResult RefreshToken([FromBody] SepesTokenClass OldSepesTokenString) 
+        {
+            IActionResult response = Unauthorized();
+            //TODO logic for verifying and then making new sepestoken.
+            //Authentication implied by [Authorize] but some content of old token may be transplanted to new token.
+            
+            var SepesToken = GenerateJSONWebToken(null ,new JwtSecurityTokenHandler().ReadJwtToken(OldSepesTokenString.idToken));
+            //Either change GenerateJSONWebToken to support either AzToken or SepeToken. Or make new function for refresh
+            response = Ok(SepesToken);
+            return response;
+        }
+
+        private string GenerateJSONWebToken(AzTokenClass AZtoken,JwtSecurityToken OldSepesToken)
+        {
+
+            if (OldSepesToken != null){
+                //Logic for bringing transplanted traits into next token
+            }
+            else if (AZtoken != null){
+                //Logic for any transplanting traits from AzToken to sepestoken
+            }
+            else {
+                //Throw exception as this code should never be reached
+                throw new ArgumentNullException("Both arguments can not be null");
+            }
+            var JwtKey = _appSetting.Key;
+            var securityKey = new SymmetricSecurityKey(System.Text.Encoding.UTF8.GetBytes(JwtKey));  //Encoding.UTF8.GetBytes(configreader["Jwt:Key"])
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256); 
+
+            var token = new JwtSecurityToken(_appSetting.Issuer,  //Issuer
+            _appSetting.Issuer,//Audience  
+            null,  
+            //TODO add in the token user so that will be usable for logging.
+            notBefore: DateTime.Now,
+            expires: DateTime.Now.AddMinutes(120),  //TODO judge how long tokens should live
+            signingCredentials: credentials);  
+  
+            return new JwtSecurityTokenHandler().WriteToken(token);
+        }
+        
+        private bool AuthenticateToken(AzTokenClass AZtoken)
+        {
+            //TODO Logic for authenticating the azure token.
+            return true; //Always returns true for testing purpose
+        }
+
+    }
+
+    public class AzTokenClass{
+        //TODO add parameters to token
+        public string Username { get; set; }
+        public string idToken { get; set; }
+        public string Expiration { get; set; }
+    }
+
+       public class SepesTokenClass{
+        //Judge what parameters are needed for just refreshing. Username and such should be in token content body
+        public string Username { get; set; }
+        public string idToken { get; set; }
+        public string Expiration { get; set; }
     }
```

Comment:
Would like to see one generic token class instead of two that are basically the same.

Reply:
They represent two different token formats. Currently contain the same content but may differ in the future. If it turns out we dont need two different classes it is trivial to combine them later.